<?xml version="1.0"?>
<launch>
    
    <arg name="num_cars" value="4" />

    <include file="$(find multi_car_localization)/launch/four_fake_cars.launch">
        <param name="num_cars" value="$(arg num_cars)" />
    </include>

    <node pkg="multi_car_localization" type="initial_pose_node.py"
        name="initial_pose_node" output="screen">
        <param name="num_cars" value="$(arg num_cars)" />
    </node>

    <node pkg="multi_car_localization" type="vicon_to_gps.py" name="vicon_to_gps" output="screen">
        <param name="num_cars" value="$(arg num_cars)" />
    </node>

    <group ns="car0">


        <node pkg="multi_car_localization" type="fake_vicon.py" name="fake_vicon" output="screen">
                <param name="num_cars" value="$(arg num_cars)" />
        </node>

    </group>

    <group ns="car1">
        <node pkg="multi_car_localization" type="fake_vicon.py" name="fake_vicon" output="screen">
                <param name="num_cars" value="$(arg num_cars)" />
        </node>

<!--         <node pkg="multi_car_localization" type="vicon_to_gps.py" name="vicon_to_gps" output="screen">
            <param name="num_cars" value="$(arg num_cars)" />
        </node> -->

    </group>

    <group ns="car2">
        <node pkg="multi_car_localization" type="fake_vicon.py" name="fake_vicon" output="screen">
                <param name="num_cars" value="$(arg num_cars)" />
        </node>

<!--         <node pkg="multi_car_localization" type="vicon_to_gps.py" name="vicon_to_gps" output="screen">
            <param name="num_cars" value="$(arg num_cars)" />
        </node> -->

    </group>

    <group ns="car3">
        <node pkg="multi_car_localization" type="fake_vicon.py" name="fake_vicon" output="screen">
                <param name="num_cars" value="$(arg num_cars)" />
        </node>

<!--         <node pkg="multi_car_localization" type="vicon_to_gps.py" name="vicon_to_gps" output="screen">
            <param name="num_cars" value="$(arg num_cars)" />
        </node> -->

    </group>

    <node pkg="tf" type="static_transform_publisher" name="map_to_gps"
        args="0 0 0 0 0 0 earth utm 100" />

    <node pkg="tf" type="static_transform_publisher" name="map_to_utm"
        args="327841.212895 4692094.36513 0 0 0 0 utm map 100" />

<!--     <node pkg="tf" type="static_transform_publisher" name="map_to_car0"
        args="0 0 0 0 0 0 map car0 100" />

    <node pkg="tf" type="static_transform_publisher" name="map_to_car1"
        args="0 0 0 0 0 0 map car1 100" />

    <node pkg="tf" type="static_transform_publisher" name="map_to_car2"
        args="0 0 0 0 0 0 map car2 100" />

    <node pkg="tf" type="static_transform_publisher" name="map_to_car3"
        args="0 0 0 0 0 0 map car3 100" /> -->

</launch>